orte/orted/pmix/pmix_server_register_fns.c

```diff
@@ -324,6 +288,43 @@ int orte_pmix_server_register_nspace(orte_job_t *jdata)
         kv->data.integer = pptr->name.vpid;
         opal_list_append(pmap, &kv->super);
 
+        /* local size - same for all on host, but not same across namespace */
+        kv = OBJ_NEW(opal_value_t);
+        kv->key = strdup(OPAL_PMIX_LOCAL_SIZE);
+        kv->type = OPAL_UINT32;
+        kv->data.uint32 = jdata->num_local_procs;
+        opal_list_append(pmap, &kv->super);
+
+        /* pass our node ID - same for all on host, but not same across namespace */
+        kv = OBJ_NEW(opal_value_t);
+        kv->key = strdup(OPAL_PMIX_NODEID);
+        kv->type = OPAL_UINT32;
+        kv->data.uint32 = node->index;
+        opal_list_append(pmap, &kv->super);
+
+        /* construct the list of peers for transmission */
+        /* - same for all on host, but not same across namespace */
+        if (NULL != list) {
+            tmp = opal_argv_join(list, ',');
+            /* pass the list of peers */
+            kv = OBJ_NEW(opal_value_t);
+            kv->key = strdup(OPAL_PMIX_LOCAL_PEERS);
+            kv->type = OPAL_STRING;
+            kv->data.string = tmp;
+            opal_list_append(pmap, &kv->super);
+        }
+        /* construct the list of cpusets for transmission */
+        /* - same for all on host, but not same across namespace */
+        if (NULL != procs) {
+            tmp = opal_argv_join(procs, ':');
+            /* pass the list of cpusets */
```

Comment:
same thing here - this data is just for the one node. If you really want the list of cpusets for the remote procs, then you'll need to regenerate that each time thru the loop.

In that case, you are probably better off looping by node instead of by proc - that way, you can generate the list and procs arrays only once, and then loop across all procs on that node and generate their "blob".

Let me know if you want help with that optimization.